### ISSUE TITLE
bug: Apisix ingress controller does not update upstream when endpoint update

### ISSUE THREAD

**Issue by damoshushu** [apache/apisix-ingress-controller#2252]
### Current Behavior

if apisix ingress controller is started before the target service, it may create an upstream with an empty node "[]".
after the endpoint is updated, apisix ingress controller does not update the upstream.

```
        {
            "createdIndex": 27,
            "key": "/apisix/upstreams/4405a246",
            "value": {
                "name": "apis_sa-front-service_80",
                "id": "4405a246",
                "type": "roundrobin",
                "nodes": [],
                "timeout": {
                    "send": 60,
                    "connect": 60,
                    "read": 60
                },
                "scheme": "http",
                "labels": {
                    "managed-by": "apisix-ingress-controller"
                },
                "desc": "Created by apisix-ingress-controller, DO NOT modify it manually"
            },
            "modifiedIndex": 27
        },
```

```
➜  ~ kubectl -n apis get ep sa-front-service
NAME               ENDPOINTS        AGE
sa-front-service   172.17.7.43:80   18m
```

### Expected Behavior

apisix ingress controller can monitor the endpoint change and update upstream accordingly

### Steps to Reproduce

1. install apisix ingress controller
2. install target service

### Environment

- APISIX Ingress controller version (run `apisix-ingress-controller version --long`)
- Kubernetes cluster version (run `kubectl version`)
- OS version if running APISIX Ingress controller in a bare-metal environment (run `uname -a`)

**Comment by sereneshikari** [comment]
I've observed a similar issue. For now, I've worked around this by setting the `resolveGranularity` to `service` so that the ingress controller doesn't need to take care of updating the upstream.

**Comment by damoshushu** [comment]
> I've observed a similar issue. For now, I've worked around this by setting the `resolveGranularity` to `service` so that the ingress controller doesn't need to take care of updating the upstream.

Hi  @sereneshikari ,
Could you please share how did you config it? I only find it on ApisixRoute config https://apisix.apache.org/docs/ingress-controller/concepts/apisix_route/#service-resolution-granularity
Can we set it on Ingress resource?

**Comment by sereneshikari** [comment]
I don't think it is possible to set this field on the `Ingress` resource. You can see the list of annotations we can use to customize settings for the `Ingress` resource [here](https://apisix.apache.org/docs/ingress-controller/concepts/annotations/). Unfortunately, none of them mentions anything about the service resolution granularity.

**Comment by damoshushu** [comment]
> I don't think it is possible to set this field on the `Ingress` resource. You can see the list of annotations we can use to customize settings for the `Ingress` resource [here](https://apisix.apache.org/docs/ingress-controller/concepts/annotations/). Unfortunately, none of them mentions anything about the service resolution granularity.

Thanks @sereneshikari , are you creating ApisixRoute directly in k8s cluster?

**Comment by sereneshikari** [comment]
> You can see the list of annotations we can use to customize settings for the Ingress resource [here](https://apisix.apache.org/docs/ingress-controller/concepts/annotations/). Unfortunately, none of them mentions anything about the service resolution granularity.

Related: https://github.com/apache/apisix-ingress-controller/pull/2181

**Comment by SuperBo** [comment]
thank @sereneshikari, I also met this issue while using cluster with spot instance.

**Comment by sereneshikari** [comment]
@SuperBo, I don't think `resolveGranularity` is applicable for the `Ingress` resource. It's only applicable for the `ApisixRoute` resource.
